internal/parser/strconv.go

```diff
@@ -120,6 +120,8 @@ func writeNextUnescapedRune(s string, builder *strings.Builder) (width int, err
 			r = '\\'
 		case '"':
 			r = '"'
+		case '$':
+			r = '$'
```

Comment:
Does this needs to check that the next rune is `{`? Otherwise it will incorrectly accept `\$` as a valid escape sequence? Or does the scanner ensure its always `${` together?

Reply:
The scanner should ensure that it is always `\${` all together so it would be difficult to make this one multiple characters.